packages/pwa-buildpack/lib/Utilities/serve.js

```diff
@@ -34,62 +29,57 @@ async function serve() {
                     ...config.section('imageOptimizing'),
                     ...config.section('imageService')
                 });
-                app.use(bestPractices());
             }
         }
     );
 
     let envPort;
     if (process.env.PORT) {
-        console.log(`PORT is set in environment: ${process.env.PORT}`);
+        prettyLogger.info(`PORT is set in environment: ${process.env.PORT}`);
         envPort = process.env.PORT;
     } else if (stagingServerSettings.port) {
-        console.log(
+        prettyLogger.info(
             `STAGING_SERVER_PORT is configured: ${stagingServerSettings.port}`
         );
         envPort = stagingServerSettings.port;
     }
 
     if (config.isProd) {
-        console.log(
+        prettyLogger.info(
             `NODE_ENV=production, will not attempt to use custom host or port`
         );
 
         if (envPort) {
             upwardServerOptions.port = envPort;
         } else {
-            console.log(`No port set. Binding to random open port`);
+            prettyLogger.warn(`No port set. Binding to random open port`);
             upwardServerOptions.port = 0;
         }
     } else {
         try {
             // don't require configureHost until you need to, since loading
             // the devcert library can have side effects.
-            const {
-                Utilities: { configureHost }
-            } = require('@magento/pwa-buildpack');
+            const configureHost = require('./configureHost');
+
             const { hostname, ports, ssl } = await configureHost(
                 Object.assign(config.section('customOrigin'), {
-                    dir: __dirname,
+                    dir: dirname,
                     interactive: false
                 })
             );
             upwardServerOptions.host = hostname;
             upwardServerOptions.https = ssl;
             upwardServerOptions.port = envPort || ports.staging || 0;
         } catch (e) {
-            console.log(
+            prettyLogger.error(
                 'Could not configure or access custom host. Using loopback...',
                 e.message
             );
         }
     }
```

Comment:
This is another place where my original implementation was buggy. This code needs to check whether `CUSTOM_ORIGIN_ENABLED` is set. It should only run `configureHost()` if the feature is enabled; otherwise it should just bind to `0.0.0.0`.

We need to wrap this entire try/catch statement in a boolean check. Can you do this?

```diff
diff --git a/packages/pwa-buildpack/lib/Utilities/serve.js b/packages/pwa-buildpack/lib/Utilities/serve.js
index 8b22a8ae5..c91110d56 100644
--- a/packages/pwa-buildpack/lib/Utilities/serve.js
+++ b/packages/pwa-buildpack/lib/Utilities/serve.js
@@ -56,25 +56,28 @@ module.exports = async function serve(dirname) {
             upwardServerOptions.port = 0;
         }
     } else {
-        try {
-            // don't require configureHost until you need to, since loading
-            // the devcert library can have side effects.
-            const configureHost = require('./configureHost');
+        const customOrigin = config.section('customOrigin');
+        if (customOrigin.enabled) {
+            try {
+                // don't require configureHost until you need to, since loading
+                // the devcert library can have side effects.
+                const configureHost = require('./configureHost');
 
-            const { hostname, ports, ssl } = await configureHost(
-                Object.assign(config.section('customOrigin'), {
-                    dir: dirname,
-                    interactive: false
-                })
-            );
-            upwardServerOptions.host = hostname;
-            upwardServerOptions.https = ssl;
-            upwardServerOptions.port = envPort || ports.staging || 0;
-        } catch (e) {
-            prettyLogger.error(
-                'Could not configure or access custom host. Using loopback...',
-                e.message
-            );
+                const { hostname, ports, ssl } = await configureHost(
+                    Object.assign(customOrigin, {
+                        dir: dirname,
+                        interactive: false
+                    })
+                );
+                upwardServerOptions.host = hostname;
+                upwardServerOptions.https = ssl;
+                upwardServerOptions.port = envPort || ports.staging || 0;
+            } catch (e) {
+                prettyLogger.error(
+                    'Could not configure or access custom host. Using loopback...',
+                    e.message
+                );
+            }
         }
     }
```